integrationtests/src/test/java/io/kroxylicious/proxy/filter/RejectingCreateTopicFilter.java

```diff
@@ -27,8 +27,10 @@ public class RejectingCreateTopicFilter implements CreateTopicsRequestFilter {
     public static final String ERROR_MESSAGE = "rejecting all topics";
     private final ForwardingStyle forwardingStyle;
     private final boolean withCloseConnection;
+    private final FilterContext constructionContext;
 
-    public RejectingCreateTopicFilter(RejectingCreateTopicFilterConfig config) {
+    public RejectingCreateTopicFilter(FilterContext constructionContext, RejectingCreateTopicFilterConfig config) {
+        this.constructionContext = constructionContext;
```

Comment:
You mean the filter holding a reference to the construction context. Yeah, we can't really prevent a filter author doing this, the context will be part of the kroxylicious-api and available in the contributor, so there's nothing to prevent them passing it through and holding a reference.

Reply:
I suppose you could make the construction context have a life-cycle and close it after construction, then throw errors if the user kept using it. But that doesn't feel so intuitive either.

Reply:
Yeah. I don't think we actually need to prevent it as you say it makes sense.

My mental model had it as something ephemeral and short lived but I don't think it needs to be